pkg/agent/globalstate.go

```diff
@@ -172,9 +146,170 @@ func (s *agentState) handleVMEventAdded(
 		status:        status,
 		vmInfoUpdated: txVMUpdate,
 	}
+	s.metrics.runnerStarts.Inc()
 	runner.Spawn(runnerCtx, rxVMUpdate)
 }
 
+// FIXME: make these timings configurable.
+const (
+	RunnerRestartMinWaitSeconds = 5
+	RunnerRestartMaxWaitSeconds = 10
+)
+
+func (s *agentState) TriggerRestartIfNecessary(runnerCtx context.Context, podName util.NamespacedName, podIP string) {
+	// Three steps:
+	//  1. Check if the Runner needs to restart. If no, we're done.
+	//  2. Wait for a random amount of time (between RunnerRestartMinWaitSeconds and RunnerRestartMaxWaitSeconds)
+	//  3. Restart the Runner (if it still should be restarted)
+
+	pod, ok := func() (*podState, bool) {
+		// note: intentionally release s.lock as soon as we're done with it, so that there's no
+		// possibility of holding onto it longer than we need.
```

Comment:
I think that comments like this really just duplicate what's going on in the code, and we should let the code speak for itself. Using a lambda for this kind of thing is fine, and it would be fine to write a specific method to do this operation (which would help with the documentation aspect, on the idea that one good method name is worth 10 comments). 

It's also the case that this comment made me think/notice that the podState is a pointer and that wile you use this lock to access the map (which is cool/good) we can still race accesses to `podState`, you solve this by using a lock on 178. If this lambda were a method (say), the code would just "look right" and your intent would be clear. Having said that, while I'm not in favor of the comment (and think that the open issue for improved lock hygine will take care of some of this), I think if you want to leave the code as it is, you should mention that the use of `podState` is governed by its own lock.